### PR TITLE
Feature/ghcr integration

### PR DESCRIPTION
**Pull request by just-at-uber**
Experimenting with integrating with ghcr for docker image uploads.

**Some caveats:**
- currently the package is not visible from the repository. You will need to go to the maintainers packages and see it there. I am unsure if switching to org if this helps at all? My understanding is GitHub is aware of this and is working on a fix (see [here](https://github.community/t/github-container-registry-ghcr-io-packages-not-appearing-in-webinterface/130077/2))
- requires a `PAT` token instead of `GITHUB_TOKEN`. github is aware of issue and working on integrating `GITHUB_TOKEN` (see [here](https://github.community/t/github-packages-not-showing-up-in-organization-packages-ui/130481/4))
- ideally needs to be setup with Uber Open Source organisation. Currently it is setup with my personal account, however if I leave the org, this pipeline will break. This may not be an issue if `GITHUB_TOKEN` is fixed.
- by default the package will be private. Once a package is published you will need to enable to public for the first time only.

## Screenshots
**just-at-uber release**
Screen Shot 2020-10-29 at 1 54 18 PM

**just-at-uber packages**
Screen Shot 2020-10-29 at 1 55 09 PM

**just-at-uber cadence/server packages**
Screen Shot 2020-10-29 at 2 00 41 PM

---------------------------------------

**What changed?**

**Why?**

**How did you test it?**

**Potential risks**

### CODE REVIEW

File: .github/workflows/docker_publish.yml [hunk 1]
```diff
@@ -40,3 +40,35 @@ jobs:
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out the repo
+        uses: actions/checkout@v2
+      - name: Push to GitHub Container Registry (server build)
+        if: "!github.event.release.prerelease"
+        uses: docker/build-push-action@v1
+        with:
+          username: ${{ github.actor }}
+          password: ${{ secrets.CADENCE_SERVER_GHCR_PAT }}
```
Comment (just-at-uber) [inline .github/workflows/docker_publish.yml:54]: Need to change back to `GITHUB_TOKEN` once github supports this.
See [here](https://github.community/t/github-packages-not-showing-up-in-organization-packages-ui/130481/4) for more information

File: .github/workflows/docker_publish.yml [hunk 2]
```diff
@@ -40,3 +40,35 @@ jobs:
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out the repo
+        uses: actions/checkout@v2
+      - name: Push to GitHub Container Registry (auto-setup build)
+        if: "!github.event.release.prerelease"
+        uses: docker/build-push-action@v1
+        with:
+          username: ${{ github.actor }}
+          password: ${{ secrets.CADENCE_SERVER_GHCR_PAT }}
```
Comment (just-at-uber) [inline .github/workflows/docker_publish.yml:70]: Need to change back to `GITHUB_TOKEN` once github supports this.
See [here](https://github.community/t/github-packages-not-showing-up-in-organization-packages-ui/130481/4) for more information

File: .github/workflows/docker_publish.yml [hunk 3]
```diff
@@ -40,3 +40,35 @@ jobs:
...
+      - name: Check out the repo
+        uses: actions/checkout@v2
+      - name: Push to GitHub Container Registry (server build)
+        if: "!github.event.release.prerelease"
+        uses: docker/build-push-action@v1
+        with:
+          username: ${{ github.actor }}
+          password: ${{ secrets.CADENCE_SERVER_GHCR_PAT }}
+          registry: ghcr.io
+          repository: just-at-uber/cadence/server
```
Comment (just-at-uber) [inline .github/workflows/docker_publish.yml:56]: Need to update repository to `uber/cadence/server` once organization has been initialized with ghcr (see [here](https://docs.github.com/en/free-pro-team@latest/packages/getting-started-with-github-container-registry/enabling-github-container-registry-for-your-organization)).

File: .github/workflows/docker_publish.yml [hunk 4]
```diff
@@ -40,3 +40,35 @@ jobs:
...
+      - name: Check out the repo
+        uses: actions/checkout@v2
+      - name: Push to GitHub Container Registry (auto-setup build)
+        if: "!github.event.release.prerelease"
+        uses: docker/build-push-action@v1
+        with:
+          username: ${{ github.actor }}
+          password: ${{ secrets.CADENCE_SERVER_GHCR_PAT }}
+          registry: ghcr.io
+          repository: just-at-uber/cadence/server
```
Comment (just-at-uber) [inline .github/workflows/docker_publish.yml:72]: Need to update repository to `uber/cadence/server` once organization has been initialized with ghcr (see [here](https://docs.github.com/en/free-pro-team@latest/packages/getting-started-with-github-container-registry/enabling-github-container-registry-for-your-organization)).

**Comment by davidporter-id-au** [comment]
🧹  but @just-at-uber  feel free to reopen if you're still interested in seeing this happen

**Pull request closed without merging**